services/QuillLMS/client/app/bundles/Teacher/components/admin_access/adminRequestModal.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react'
+
+import { requestPost, } from '../../../../modules/request'
+import { smallWhiteCheckIcon, indeterminateCheckIcon, Input, } from '../../../Shared/index'
+
+const AdminRequestModal = ({ onSuccess, schoolAdmins, closeModal, }) => {
+  const [selectedAdminIds, setSelectedAdminIds] = React.useState([])
+  const [reason, setReason] = React.useState('')
+
+  function handleChangeReason(e) { setReason(e.target.value) }
+
+  function selectAdmin(id) {
+    const newAdminIds = Array.from(new Set(selectedAdminIds.concat([id])))
+    setSelectedAdminIds(newAdminIds)
+  }
+
+  function unselectAdmin(id) {
+    const newAdminIds = selectedAdminIds.filter(k => k !== id)
+    setSelectedAdminIds(newAdminIds)
+  }
+
+  function renderCheckbox(admin) {
+    const {id, name, } = admin
+
+    if (selectedAdminIds.includes(id)) {
+      return (
+        <button aria-label={`Unselect ${name}`} className="focus-on-light quill-checkbox selected" onClick={() => unselectAdmin(id)} type="button">
+          <img alt={smallWhiteCheckIcon.alt} src={smallWhiteCheckIcon.src} />
+        </button>
+      )
+    }
+
+    return <button aria-label={`Select ${name}`} className="focus-on-light quill-checkbox unselected" onClick={() => selectAdmin(id)} type="button" />
+  }
+
+  function handleClickSendRequest() {
+    requestPost(`${process.env.DEFAULT_URL}/admin_access/request_upgrade_to_admin_from_existing_admins`, {
+      admin_ids: selectedAdminIds,
+      reason
+    }, () => {
+      onSuccess('Your request has been sent')
+    })
+  }
+
+  function selectAllAdminIds() { setSelectedAdminIds(schoolAdmins.map(sa => sa.id)) }
+
+  function unselectAllAdminIds() { setSelectedAdminIds([]) }
+
+  function renderTopLevelCheckbox() {
+    if (selectedAdminIds.length === 0) {
+      return <button aria-label="Select all admins" className="focus-on-light quill-checkbox unselected" onClick={selectAllAdminIds} type="button" />
+    } else if (schoolAdmins.length === selectedAdminIds.length) {
+      return (
+        <button aria-label="Unselect all admins" className="focus-on-light quill-checkbox selected" onClick={unselectAllAdminIds} type="button">
+          <img alt={smallWhiteCheckIcon.alt} src={smallWhiteCheckIcon.src} />
+        </button>
+      )
+    } else {
+      return (
+        <button aria-label="Unselect all admins" className="focus-on-light quill-checkbox selected" onClick={unselectAllAdminIds} type="button">
+          <img alt={indeterminateCheckIcon.alt} src={indeterminateCheckIcon.src} />
+        </button>
+      )
+    }
+  }
+
+  const adminTable = (
+    <div className="admin-selection-table">
+      <div className="admin-selection-row top-row">
+        {renderTopLevelCheckbox()}
+        <div>Select all</div>
```

Comment:
No - it isn't really a paragraph either. The most accurate HTML would be `label`, but the checkbox already has an `aria-label` so that isn't necessary.